client/components/search/index.js

```diff
@@ -71,55 +78,111 @@ class Search extends Component {
+		const { value = '', isActive } = this.state;
 		const aria = {
 			'aria-labelledby': this.props[ 'aria-labelledby' ],
 			'aria-label': this.props[ 'aria-label' ],
 		};
 		return (
 			<div className="woocommerce-search">
+				<label htmlFor={ `search-input-${ instanceId }` } className="screen-reader-text">
+					{ placeholder || __( 'Add item', 'wc-admin' ) }
+				</label>
 				<Gridicon className="woocommerce-search__icon" icon="search" size={ 18 } />
 				<Autocomplete completer={ autocompleter } onSelect={ this.selectResult }>
-					{ ( { listBoxId, activeId, onChange } ) => (
-						<input
-							type="search"
-							value={ value }
-							placeholder={ placeholder }
-							className="woocommerce-search__input"
-							onChange={ this.updateSearch( onChange ) }
-							aria-owns={ listBoxId }
-							aria-activedescendant={ activeId }
-							{ ...aria }
-						/>
-					) }
-				</Autocomplete>
-				{ selected.length ? (
-					<div className="woocommerce-search__token-list">
-						{ selected.map( ( item, i ) => {
-							const screenReaderLabel = sprintf(
-								__( '%1$s (%2$s of %3$s)', 'wc-admin' ),
-								item.label,
-								i,
-								selected.length
-							);
-							return (
-								<Tag
-									key={ item.id }
-									id={ item.id }
-									label={ item.label }
-									remove={ this.removeResult }
-									removeLabel={ __( 'Remove product', 'wc-admin' ) }
-									screenReaderLabel={ screenReaderLabel }
+					{ ( { listBoxId, activeId, onChange } ) =>
+						// Disable reason for no-static-element-interactions: There is no appropriate role
+						// which describes the input container intended accessible usability.
+						// Disable reason for click-events-have-key-events: The div below visually
+						// simulates an input field. Its child input is the actual input and
+						// responds accordingly to all keyboard events, but click events need to be
+						// passed onto the child input.
+						/* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */
+						inlineTags ? (
+							<div
+								className={ classnames( 'woocommerce-search__inline-container', {
+									'is-active': isActive,
+								} ) }
+								onClick={ () => {
+									this.input.current.focus();
+								} }
+							>
+								{ this.renderTags() }
+								<input
+									ref={ this.input }
+									type="text"
+									size={
+										( ( value.length === 0 && placeholder && placeholder.length ) ||
+											value.length ) + 1
+									}
+									id={ `search-input-${ instanceId }` }
+									value={ value }
+									placeholder={ ( selected.length === 0 && placeholder ) || '' }
+									className="woocommerce-search__inline-input"
+									onChange={ this.updateSearch( onChange ) }
+									aria-owns={ listBoxId }
+									aria-activedescendant={ activeId }
+									onFocus={ this.onFocus }
+									onBlur={ this.onBlur }
+									{ ...aria }
```

Comment:
It might be good to add an `aria-describedby` div that said "Move [forward/backward] for selected items" or something like that, so screen reader users know where these show up